### PR TITLE
Clean fetched data (reformat certain values and remove unused data before setting to state)

### LINKED ISSUE #42: Clean fetched data

### PR DESCRIPTION
**Pull request by alexmkio**
### Expected Behavior
When movies array is fetched do not include the avg.rating, backdrop_path, and release_date properties in the data being set to state.
When a particular movie is fetched reformat average_rating, revenue, and budget AND do not include the id or poster_path properties in the data being set to state.

### Issues Resolved
Remove unused fetched data before assigning it to our data model (setting state).

### Code Reviewers
@LKessell 

### Professionalism Rubric
- README concisely communicates the team’s individual and joint learning goals, the evolution of the project, and team member reflections while using good formatting to enhance readability
- README links to all user GitHub profiles and any applicable repos/deployed sites
- Team uses a rebase workflow
- Git commits are atomic, with concise and precise descriptions of the change made
- PRs have full, consistent descriptions
- Team members do consistent, thorough, meaningful code reviews of PRs, which prompt updates and changes made to that PR before merging
- Evolution of the project (decisions made, etc) are fully and clearly documented in the git history and PRs
- When the project is run locally, the terminal shows no errors or warnings

### React Rubric
- A consistent, modular file structure is used
- A clear understanding of class components vs function components is demonstrated
- Only the data that a child component needs is passed down as props
- Logic is kept out of return statements; return statements are as readable as possible, only communicating what will be displayed
- Fetch calls have been refactored to be reusable for multiple queries
- Frontend data (state) always matches the backend data
- Data fetched from API is run through a cleaning function (which is defined in a separate utilities file)
- Implements excellent error handling if movie database server is down or fetch fails (this includes loading images as well as error messages on the frontend)

### Testing Rubric
- Team has successfully achieved 90%+ test coverage of all components
- All async functionality is stubbed and tested
- Async tests cover happy & sad paths
- All application views are tested
- All user flows are tested

### Routing Rubric
- Application has been refactored to use Router without leaving artifacts of the prior code (no odd workarounds remaining)
- Use of Router shows developer empathy (readability, maintainability)
- Application uses React Router components and does not manipulate the history object
- A 404 page handles undefined routes
- UX is excellent; routes are consistent and navigation is clear

### Mod1 Best Practices
- All functions strictly adhere to the [Single Responsibility Principle (SRP)](https://dev.to/skill_pathway/single-responsibility-principle-for-dummies-59gb) and are all 10 lines of code or less
- Uses logical operators instead of nesting conditional statements
- Conditionals are written in a way that demonstrates an understanding of [https://frontend.turing.io/lessons/module-1/js-truthy-falsy-expressions.html](truthy/falsy values). For example, the developer avoids using logic like <code class="language-plaintext highlighter-rouge">=== true</code> and <code class="language-plaintext highlighter-rouge">!== 0</code>.
- Functions are [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and observe [SRP](https://dev.to/skill_pathway/single-responsibility-principle-for-dummies-59gb) - and most are around 10 lines of code or less
- Most functions and variables are semantically and concisely named. Functions should start with a verb, array names should be plural, etc. Function and variable names describe their role in the program. Examples: The name of the data type should not ever be in a variable name (ex: “petArray”); the name itself should be clear enough to indicate the type of data it holds (ex: “allPets” or “pets”).
- An event handler should not have the word “handler” in the name (ex: “clickHandler”); the name should indicate the handler’s purpose (ex: “addNewPet”).
- Uses parameters and arguments to craft short and reusable functions. There are no parameters or arguments in functions that are unused.
- All console logs, debuggers and comments are removed from code before submitting.

### CODE REVIEW

File: src/components/app/App.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: [],
+      movies: null,
```
Comment (LKessell) [inline src/components/app/App.js:14]: Nice, I feel like null does make more sense here.

File: src/components/app/App.js [hunk 2]
```diff
@@ -24,16 +24,37 @@ class App extends Component {
...
         let fetchedMovies = await getApiData(endPoint)
-        this.setState({movies: fetchedMovies.movies})
+        this.setState({movies: this.cleanData(endPoint, fetchedMovies)})
       }
     } catch (e) {
       this.setState({errorCode: e.message});
     }
   }
 
+  cleanData = (urlEndPoint, data) => {
```
Comment (LKessell) [inline src/components/app/App.js]: This function looks great, but the spec suggests that we put it in a separate `utilities` file. Can we separate it out, then import the function?

**Review by LKessell — CHANGES_REQUESTED** [review]

File: src/components/app/App.js (hunk 2 above)
Comment (alexmkio) [reply src/components/app/App.js]: you're right! I missed that detail, but just pushed a commit to resolve this.

Comment (LKessell) [reply src/components/app/App.js]: Thanks for doing that, looks perfect now!

**Pull request merged**